File: Examples/Getting-Started/Search-Inventory/Python/search_inventory.py
import os

from netld_example_client import NetLDClient, print_devices


client = NetLDClient.from_env()
client.login()

page_data = client.search_inventory(
    networks=[os.environ.get("NETLD_NETWORK", "Default")],
    schemes=os.environ.get("NETLD_SEARCH_SCHEME", "ipAddress"),
    queries=os.environ.get("NETLD_SEARCH_QUERY", "10.95.1.0/24"),
)

print_devices(page_data)

File: Examples/Getting-Started/Search-Inventory/nodeJS/search-inventory.mjs
import "dotenv/config";

import { NetLDClient, printDevices } from "./netld-example-client.mjs";

const client = NetLDClient.fromEnv();
await client.login();

const pageData = await client.searchInventory({
  networks: [process.env.NETLD_NETWORK || "Default"],
  schemes: process.env.NETLD_SEARCH_SCHEME || "ipAddress",
  queries: process.env.NETLD_SEARCH_QUERY || "10.95.1.0/24",
});

printDevices(pageData);
